src/main/java/io/stargate/sgv2/jsonapi/service/updater/DocumentUpdater.java

```diff
@@ -82,7 +82,7 @@ private boolean replace(ObjectNode docToUpdate, boolean docInserted) {
     ObjectNode compareDoc = docToUpdate.deepCopy();
     JsonNode idNode = compareDoc.remove(DocumentConstants.Fields.DOC_ID);
     // The replace document cannot specify an _id value that differs from the replaced document.
-    if (replaceDocumentId != null) {
+    if (replaceDocumentId != null && idNode != null) {
```

Comment:
This avoids NPE for case where no document exists (hence no `idNode`). Or rather, we have "empty document".